src/windows/volume.cc

```diff
@@ -65,6 +65,41 @@ HRESULT drivelist::volume::GetReadOnlyFlag(const wchar_t letter, BOOL *out) {
   return result;
 }
 
+HRESULT drivelist::volume::HasFileSystem(const wchar_t letter, BOOL *out) {
```

Comment:
Surely checking the partition type is something very different to a volume having a filesystem or not?
Consider creating a FAT32 partition on a disk (which updates the MBR) but then not formatting it. The volume has a valid partition type, but it doesn't have any filesystem yet...

IMHO either this function isn't doing what you think it is, or it should be renamed to something else...